packages/components/src/test/tree/controlled/UseHierarchyLevelFiltering.test.ts

```diff
@@ -208,4 +209,41 @@ describe("useHierarchyLevelFiltering", () => {
     result.current.clearFilter(parentNode.item);
     expect(modelSource.getModel().getNode(childNode.id)).to.be.undefined;
   });
+
+  it("clears filter unsubscribes from observable created by `applyFilter`", () => {
+    const applyFilterActionSubject = new Subject<TreeNodeLoadResult>();
+    const clearFilterActionSubject = new Subject<TreeNodeLoadResult>();
+    let isFilterCalled = false;
+    nodeLoaderMock.setup((x) => x.loadNode(moq.It.isAny(), 0)).returns(() => isFilterCalled ? clearFilterActionSubject : applyFilterActionSubject);
+
+    const node = createTreeModelInput(undefined, { filtering: { descriptor: createTestContentDescriptor({ fields: [] }) } });
+    modelSource.modifyModel((model) => {
+      model.setChildren(undefined, [node], 0);
+    });
+
+    const { result } = renderHook(useHierarchyLevelFiltering, { initialProps: { modelSource, nodeLoader: nodeLoaderMock.object } });
+
+    result.current.applyFilter(node.item, filterInfo);
+    isFilterCalled = true;
+    result.current.clearFilter(node.item);
+
+    expect(applyFilterActionSubject.observers.length).to.be.eq(0);
```

Comment:
I think there should be an expect statement after `applyFilter` to check if there are any observers.